Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-volumegroup-snapshot.md

```diff
@@ -0,0 +1,100 @@
+
+- Add the required args to the snapshot controller for volumegroup snapshot support. See [here](https://github.com/kubernetes-csi/external-snapshotter/tree/master#volume-group-snapshot-support) for more details.
+
+- We also need a `VolumeGroupSnapshotClass` for volumegroup snapshot to work. The purpose of a `VolumeGroupSnapshotClass` is
+defined in [the kubernetes
+documentation](https://kubernetes.io/blog/2023/05/08/kubernetes-1-27-volume-group-snapshot-alpha/).
+In short, as the documentation describes it:
+
+!!! info
+    Created by cluster administrators to describe how volume group snapshots
+    should be created. including the driver information, the deletion policy, etc.
+
+## CephFS Snapshots
+
+### VolumeGroupSnapshotClass
+
+In [VolumeGroupSnapshotClass](https://github.com/rook/rook/tree/master/deploy/examples/csi/cephfs/groupsnapshotclass.yaml),
+the `csi.storage.k8s.io/group-snapshotter-secret-name` parameter should reference the
+name of the secret created for the cephfsplugin.
+
+In the `VolumeGroupSnapshotClass`, update the value of the `clusterID` field to match the namespace
+that Rook is running in. When Ceph CSI is deployed by Rook, the operator will automatically
+maintain a configmap whose contents will match this key. By default this is
+"rook-ceph".
+
+```console
+kubectl create -f deploy/examples/csi/cephfs/groupsnapshotclass.yaml
+```
+
+### VolumeGroupSnapshot
+
+In [VolumeGroupSnapshot](https://github.com/rook/rook/tree/master/deploy/examples/csi/cephfs/groupsnapshot.yaml),
+`volumeGroupSnapshotClassName` should be the name of the `VolumeGroupSnapshotClass`
+previously created. The labels inside `matchLabels` should be the present on the
+PVC which is already created by the CephFS CSI driver.
+
+```console
+kubectl create -f deploy/examples/csi/cephfs/groupsnapshot.yaml
+```
+
+### Verify CephFS GroupSnapshot Creation
+
+```console
+$ kubectl get volumegroupsnapshotclass
+NAME                              DRIVER                          DELETIONPOLICY   AGE
+csi-cephfsplugin-groupsnapclass   rook-ceph.cephfs.csi.ceph.com   Delete           21m
+```
+
+```console
+$ kubectl get volumegroupsnapshot
+NAME                       READYTOUSE   VOLUMEGROUPSNAPSHOTCLASS          VOLUMEGROUPSNAPSHOTCONTENT                              CREATIONTIME   AGE
+cephfs-pvc-groupsnapshot   true         csi-cephfsplugin-groupsnapclass   groupsnapcontent-77ef7c5b-93ae-4ec3-b880-fa70740fb681   2m27s          2m29s
+```
+
+The snapshot will be ready to restore to a new PVC when `READYTOUSE` field of the
+`volumegroupsnapshot` is set to true.
+
+### Restore the CephFS groupsnapshot to a new PVC
+
+In
+[pvc-group-restore](https://github.com/rook/rook/tree/master/deploy/examples/csi/cephfs/pvc-restoregroup.yaml),
+`dataSource` should be the name of the `VolumeGroupSnapshot` previously
+created. The `dataSource` kind should be the `VolumeSnapshot`.
```

Comment:
This seems to be wrong we need to first identify the volumesnapshot name(s) from the volumegroupsnapshot object and use that for restore operation.

Reply:
Right, will update :)